### DISCUSSION TITLE
Onedrive

### DISCUSSION THREAD

**Discussion by compie67** [leokhoa/laragon#654] [Q&A]
Hi there
i want to use Laragon for students. 
from usb it work great.
But i want to use this from the student OneDrive so that they can use the school shared computers
usbwebserver got this working, but want to use Laragon

Is this in anyway posible?

regards
Eric

**Comment by leokhoa** [comment]
@compie67: Laragon is portable so if it works on USB, it should work with OneDrive.

**Reply by compie67** [reply]
Thats is my thought are also.
But got this
![image](https://github.com/leokhoa/laragon/assets/6383681/0f008b3c-972d-444f-8207-2102866760fd)

And sample Larogon not working directory is this
**C:\Users\Eric Redegeld\OneDrive - Stichting VO Amsterdam Zuid\laragon**

And the USBwebserver working is this

**C:\Users\Eric Redegeld\OneDrive - Stichting VO Amsterdam Zuid\USBWebserver v8_en**

reason for largon is that the usbwebserver is outdated

**Comment by leokhoa** [comment]
@compie67 : The error is relevant. You should not use SPACE in path to avoid unwanted issues later on.
There are 2 options:
1. Store OneDrive in another path with no space
2. Create a soft link to current OneDrive

**Comment by Pen-y-Fan** [comment]
I would recommend Laragon be installed on each computer, as PHP has dependencies, such as VS15 C++ runtimes, which Laragon automatically installs.

The students can then configure Laragon to serve the www directory from their one drive. See The document and data root in preferences: https://laragon.org/docs/screenshots.html#Preferences 

As leo already suggested this could be mapped to a specific drive, e.g. P: or X: to keep to data and projects per student.
